h2o-core/src/main/java/water/api/Handler2.java

```diff
@@ -0,0 +1,26 @@
+package water.api;
+
+/**
+ * Common interface for <s>all</s> some REST endpoint handlers.
+ *
+ * @param <IS> input schema class
+ * @param <OS> output schema class
+ */
+public abstract class Handler2<IS extends Schema, OS extends Schema> extends Handler {
```

Comment:
I'm wary about `AbstractHandler`, because it is weird to have `AbstractHandler` inheriting from `Handler`, not the other way around. I can live with `RestApiHandler`, with a tacit assumption that it might one day become the primary way to create new endpoints.